packages/core/src/chains/ton/sdkTon/utils.ts

```
@@ -0,0 +1,45 @@
+        return Buffer.from([0]);
+    }
+
+    // Split & validate labels
+    const labelsAscii = norm.split('.').map((lbl) => {
+        if (lbl.length === 0) {
+            throw new Error('Empty label ("..") not allowed');
+        }
+        // IDN: convert Unicode → punycode ASCII (xn--…)
+        const ascii = toASCII(lbl);
+        // Disallow bytes 0x00–0x20 and label > 63 chars (classic DNS rule)
+        if (ascii.length > 63 || /[\x00-\x20]/.test(ascii)) {
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)